lib/src/main/scala/spinal/lib/bus/regif/BusIfBase.scala

```diff
@@ -1,37 +1,50 @@
 package spinal.lib.bus.regif
 
 import spinal.core._
-import spinal.lib.BigIntRicher
-import spinal.lib.bus.amba3.apb._
-import spinal.lib.bus.misc.SizeMapping
+import spinal.core.fiber.Handle
+import spinal.lib.{BigIntRicher, WhenBuilder}
 
 import language.experimental.macros
 import scala.collection.mutable.ListBuffer
 
 trait BusIfBase extends Area{
+  val busDataWidth: Int
+  val busAddrWidth: Int
+
   val askWrite: Bool
   val askRead: Bool
   val doWrite: Bool
   val doRead: Bool
+  val cg_en: Bool = True
+  val NS: Bool = False //NS(Non-Secure Access flag bit)
+
+  val bus_rderr: Bool
+  val bus_rdata: Bits
+  val reg_rderr: Bool
+  val reg_rdata: Bits
 
-  val readData: Bits
```

Comment:
This was one of the changes I was referring to when I said this PR is not backwards compatible. This change breaks ever custom bus implementation a user of the library has